extensions/desktop/common/desktop-exports

```diff
@@ -173,22 +183,18 @@ for ((i = 0; i < ${#XDG_SPECIAL_DIRS_PATHS[@]}; i++)); do
 done
 
 # If needs XDG update and xdg-user-dirs-update exists in $PATH, run it
-if [ $needs_xdg_update = true ] && command -v xdg-user-dirs-update >/dev/null; then
+if [ "$needs_xdg_update" = true ] && command -v xdg-user-dirs-update >/dev/null; then
   xdg-user-dirs-update
   update_xdg_dirs_values
 fi
 
 # Create links for user-dirs.dirs
-if [ $needs_xdg_links = true ]; then
+if [ "$needs_xdg_links" = true ]; then
   for ((i = 0; i < ${#XDG_SPECIAL_DIRS_PATHS[@]}; i++)); do
-    b="$(realpath "${XDG_SPECIAL_DIRS_PATHS[$i]}" --relative-to="$HOME")"
+    b=$(realpath "${XDG_SPECIAL_DIRS_PATHS[$i]}" --relative-to="$HOME")
     if [ -e "$REALHOME/$b" ]; then
-      if [ -d "$HOME/$b" ]; then
-        rmdir "$HOME/$b" 2> /dev/null
-      fi
-      if [ ! -e "$HOME/$b" ]; then
-        ln -s "$REALHOME/$b" "$HOME/$b"
-      fi
+      [ -d "$HOME/$b" ] && rmdir "$HOME/$b" 2> /dev/null
+      [ ! -e "$HOME/$b" ] && ln -s "$REALHOME/$b" "$HOME/$b"
```

Comment:
Should there be a warning or something if we can't link it because it's non-empty?

Reply:
that might break on updates or new xdg links, let's leave that as a future improvement prior discussion with the desktop team

Reply:
Sounds good